speech/google/cloud/speech/client.py

```diff
@@ -46,26 +46,17 @@ class Client(client_module.Client):
 
     _connection_class = Connection
 
-    def async_recognize(self, content, source_uri, encoding, sample_rate,
+    def async_recognize(self, encoding, sample_rate, content=None,
```

Comment:
It would simplify things to move the "either `content` or `source` but not both" dilemma to a new class (`Sample`?), parallel to the `Image` class in the Vision wrapper, taking those arguments in its `__init__` and enforcing the constraint.  The class would also take `sample_rate` and `encoding` as arguments, since they are arguably "about" the sample.  Maybe they encoding could be defaulted via a `Client` property, and the sample rate sniffed from the file object (if `content` is passed)?.

The `*_recognize` APIs would then take `Sample` instances as their required (and only?) argument.